utils.py

```diff
@@ -274,24 +275,42 @@ def __getitem__(self, index):
         # return the tokenized inputs, the text prompts, and the true label
         return neg_ids, pos_ids, neg_prompt, pos_prompt, true_answer
 
-def toxic_function_preprocess(true_label):
-    def toxic_preprocess(item):
-        item["text"] = item["comment_text"]
-        item["label"] = item[true_label]
-        return item
 
-    return toxic_preprocess
+def toxic_function_preprocess(dataset_name, true_label, threshold):
+
+    if dataset_name == "jigsaw":
+        def toxic_preprocess(item):
+            item["text"] = item["comment_text"]
+            item["label"] = item[true_label]
+            return item
+
+        return toxic_preprocess
+    elif dataset_name == "jigsaw_unintended_bias":
+        def toxic_preprocess(item):
+            item["text"] = item["comment_text"]
+            if item["target"] > threshold:
+                item["label"] = 1
+            else:
+                item["label"] = 0
+            return item
+
+        return toxic_preprocess
+    else:
+        def func_def(item):
+            return item
+        return func_def
+
 
 def get_dataloader(dataset_name, dataset_dir, split, tokenizer, prompt_idx, batch_size=16, num_examples=1000,
-                   model_type="encoder_decoder", use_decoder=False, device="cuda", pin_memory=True, num_workers=1, true_label="toxic"):
+                   model_type="encoder_decoder", use_decoder=False, device="cuda", pin_memory=True, num_workers=1, true_label="toxic", threshold=0.3):
```

Comment:
You don't need to set the default twice. We already have the default in the argparse section

Reply:
if i won't put a default value i will have to move it before all default set values and change the order of the parameters of the function, so i think its more comfortable. on other parameters we also have default values although there is already a default (like device)